src/DependencyInjection/Configuration.php

```diff
@@ -148,4 +149,21 @@ private function createMockNode()
 
         return $node;
     }
+
+    private function createToleranceNode()
+    {
+        $treeBuilder = new TreeBuilder();
+        $node = $treeBuilder->root('tolerance');
+
+        $node
+            ->canBeEnabled()
+            ->children()
+                ->scalarNode('retry')->defaultValue(2)->end()
```

Comment:
The sense of the `retry` config was to provide a simple configuration that fits 90% of the use-cases.
But I agree with you that using `$options` is a better idea.

By the way, https://github.com/Tolerance/Tolerance/pull/86 has been merged and @sroze would like to move the middleware to the tolerance repository.